src/components/gridlines.ts

```diff
@@ -1,37 +1,119 @@
 namespace Plottable.Components {
+
+  export type ILinePosition = "center" | "after";
+  type ScaleDomainValues = (string | number)[];
+
   export class Gridlines extends Component {
-    private _xScale: QuantitativeScale<any>;
-    private _yScale: QuantitativeScale<any>;
+    private static validateScale = (scale: any, which: string) => {
+      if (scale != null &&
+          ! QuantitativeScale.prototype.isPrototypeOf(scale) &&
+          ! Plottable.Scales.Category.prototype.isPrototypeOf(scale)) {
+        throw new Error(which + " needs to inherit from Scale.QuantitativeScale or Scales.Category.");
+      }
+    };
+
+    private _xScale: Scale<any, number>;
+    private _yScale: Scale<any, number>;
     private _xLinesContainer: d3.Selection<void>;
     private _yLinesContainer: d3.Selection<void>;
+    private _xTicks: () => ScaleDomainValues;
+    private _yTicks: () => ScaleDomainValues;
 
-    private _renderCallback: ScaleCallback<QuantitativeScale<any>>;
+    /**
+     */
+    private _xLinePosition: ILinePosition;
+    private _yLinePosition: ILinePosition;
 
+    private _renderCallback: ScaleCallback<Scale<any, number>>;
     /**
      * @constructor
-     * @param {QuantitativeScale} xScale The scale to base the x gridlines on. Pass null if no gridlines are desired.
-     * @param {QuantitativeScale} yScale The scale to base the y gridlines on. Pass null if no gridlines are desired.
+     *
+     * @param {Scale<any, number>} xScale The scale to base the x
+     * gridlines on. Can be a category or numeric scale. Pass null if
+     * no gridlines are desired.
+     *
+     * @param {Scale<any, number>} yScale The scale to base the y
+     * gridlines on. Can be a category or numeric scale. Pass null if
+     * no gridlines are desired.
      */
-    constructor(xScale: QuantitativeScale<any>, yScale: QuantitativeScale<any>) {
-      if (xScale != null && !(QuantitativeScale.prototype.isPrototypeOf(xScale))) {
-        throw new Error("xScale needs to inherit from Scale.QuantitativeScale");
-      }
-      if (yScale != null && !(QuantitativeScale.prototype.isPrototypeOf(yScale))) {
-        throw new Error("yScale needs to inherit from Scale.QuantitativeScale");
-      }
+    constructor(
+        xScale: QuantitativeScale<any> | Plottable.Scales.Category,
+        yScale: QuantitativeScale<any> | Plottable.Scales.Category) {
+
+      Gridlines.validateScale(xScale, "xScale");
+      Gridlines.validateScale(yScale, "yScale");
```

Comment:
I think that plottable has added various checks for when its used outside of a typescript environment to attempt to enforce our contracts for vanilla javascript users.